src/main/java/org/opensearch/security/dlic/rest/support/Utils.java

```diff
@@ -228,22 +229,22 @@ public static String hash(final char[] clearTextPassword) {
      * @return new list of API routes prefixed with _opendistro... and _plugins...
      *Total number of routes is expanded as twice as the number of routes passed in
      */
-    public static List<Route> addRoutesPrefix(List<Route> routes){
-        return addRoutesPrefix(routes, "/_opendistro/_security/api", "/_plugins/_security/api");
+    public static List<ReplacedRoute> addRoutesPrefix(List<Route> routes){
+        return addRoutesPrefix(routes, "/_plugins/_security/api", "/_opendistro/_security/api");
     }
 
     /**
      * Add customized prefix(_opendistro... and _plugins...)to API rest routes
      * @param routes routes
-     * @param prefixes all api prefix
+     * @param newPrefix new prefix
+     * @param oldPrefix old prefix
      * @return new list of API routes prefixed with the strings listed in prefixes
      * Total number of routes will be expanded len(prefixes) as much comparing to the list passed in
      */
-    public static List<Route> addRoutesPrefix(List<Route> routes, final String... prefixes){
+    public static List<ReplacedRoute> addRoutesPrefix(List<Route> routes, final String newPrefix, final String oldPrefix){
```

Comment:
This method must be part of OS core as it is not specific for the security plugin.

Reply:
If I remember correctly, we discussed it with DB a while ago during the API renaming. So I think it should be done in another PR.

Reply:
Yes, it needs to be done in another PR. Please have the PR open against opensearch repo. Once it is approved and merged, we can do changes in the plugin repo.

Reply:
Do you mean we can not merge this until the other PR is done?

Reply:
Yes, it will have the dependency on it.

Reply:
Feel free to open a PR in OpenSearch. Looks like it would definitely help other consumers as well.
We can take iterative approach here. 

This PR doesn't have to be blocked on that. 
1. Lets push this change as it has value.
2. @afazel can you work on upstreaming this change to OpenSearch?

Reply:
Sure @saratvemulapalli , I'll open another PR for it soon.

Reply:
As @afazel and @saratvemulapalli discussed, there will be future PR for opensearch core that provide such functionality and current PR can be unblocked for allowing things to work iteratively. 

Resolving this conversation

Reply:
OpenSearch would probably not accept anything other than a critical bug fix into 1.0 branch right now, but I believe we need this change in security 1.0, so good call

Reply:
This is a trivial change in OS and it will benefit many others plugins as well. I don't see why we need to make code changes in the security plugin first and then do another round of code changes in the security plugin once the OS core provides the missing functionality. This PR can wait till the required functionality is implemented by the core.

Reply:
You can go and PR this change on `main` in OpenSearch, but it won't get backported to 1.0 because it's not a critical fix from the core perspective, it's a feature. 

However, you have a bug in the security plugin where routes aren't properly deprecated. This feels like a 1.0 issue to me. Are you saying it doesn't need to be fixed for 1.0?

Reply:
If OS only accepts critical bug fixes at this point, why security plugin should accept anything other than critical bug fixes as well? I would not consider missing warning messages in the log as a critical bug fix, so no, it does not need to be fixed in 1.0.

Reply:
This call should be made by owners of the security plugin.

Reply:
I published a PR in OpenSearch to add `addRoutesPrefix()` to `RestHandler`: https://github.com/opensearch-project/OpenSearch/pull/947

Reply:
Resolving this command as PR #947 has merged into OpenSearch